### PR TITLE
docs: generate documentation with typedoc

### PR DESCRIPTION
**Pull request by natenichols**
## High Level Overview of Change
Generate documentation with typedoc.

### Context of Change
Removes `doctoc` (which is what was giving us vulnerability warnings in npm) in favor of `typedoc`. Rewrites the `npm run docgen` command. Generates html to the `./docs` directory.

Screen Shot 2021-09-09 at 4 32 52 PM

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [x] Documentation Updates
- [ ] Release

### CODE REVIEW

**Comment by mvadari** [comment]
Can you add a screenshot of what one of the pages looks like?

File: docs/interfaces/FormattedOrderCancellation.html [hunk 1]
```diff
@@ -0,0 +1,163 @@
+<!doctype html>
```
Comment (mvadari) [inline docs/interfaces/FormattedOrderCancellation.html:1]: any way we can avoid docgen for stuff that's going to be deleted soon?

**Comment by mvadari** [comment]
What does one of the model pages look like?

Also, are we going to add docs for methods at some point?

File: test/utils/computeLedgerHash.ts [hunk 2]
```diff
@@ -45,6 +49,10 @@ describe('computeLedgerHash', function () {
     try {
       hash = computeLedgerHeaderHash(ledger, { computeTreeHashes: true })
     } catch (error) {
+      if (!(error instanceof Error)) {
+        throw error
+      }
+
```
Comment (JST5000) [inline test/utils/computeLedgerHash.ts:55]: Not sure if you meant for these changes to be in this PR as well

Comment (JST5000) [reply test/utils/computeLedgerHash.ts:55]: Nevermind, these will probably be resolved when you merge

**Review by JST5000 — APPROVED** [review]
Looks good to me. It'll probably be easier to give feedback on the live page than the code diff for something like this :)

**Comment by mvadari** [comment]
> Looks good to me. It'll probably be easier to give feedback on the live page than the code diff for something like this :)

@JST5000 you can pull down the branch and look at the HTML in your browser

**Comment by natenichols** [comment]
Documentation is now up on [my fork of the repo] (https://natenichols.github.io/ripple-lib/classes/Client.html)

**Comment by mvadari** [comment]
Notes from reviewing the HTML in my browser:
* Why does this still have the old README? 
* Is there any way to group the docs, like how readthedocs does it? Where you can have a "folder"/tab for models, another for utils, etc.

**Comment by natenichols** [comment]
This is using the old README.md because the original commit was before that was merged in 2.0

I dont know how to group the docs, lemme look into it.

The names are for the same reason as the README.md

**Comment by natenichols** [comment]
Once merged into 2.0, we'll need to re-generate the docs and it should be up to date.

**Review by mukulljangid — APPROVED** [review]
LGTM!

**Review by mvadari — APPROVED** [review]

**Pull request merged**